pkg/reconcilers/deploymentconfig.go

```diff
@@ -57,6 +57,77 @@ func GenericBackendMutators() []DCMutateFn {
 	}
 }
 
+func DeploymentConfigArgsMutator(_, existing *appsv1.DeploymentConfig) (bool, error) {
+	update := true
+	existing.Spec.Template.Spec.Containers[0].Args = []string{"bin/3scale_backend", "-s", "falcon", "start", "-e", "production", "-p", "3000", "-x", "/dev/stdout"}
+	return update, nil
+}
+
+func DeploymentConfigEnvMutator(desired, existing *appsv1.DeploymentConfig) (bool, error) {
+	update := false
+
+	// Always set env var CONFIG_REDIS_ASYNC to 1 this logic is only hit when you don't have logical redis db
+	for envId, envVar := range existing.Spec.Template.Spec.Containers[0].Env {
+		if envVar.Name == "LISTENER_WORKERS" {
+			if envVar.Value == "0" {
+				existing.Spec.Template.Spec.Containers[0].Env[envId].Value = "1"
+				update = true
+				return update, nil
+			}
+		}
+		if envVar.Name == "CONFIG_REDIS_ASYNC" {
+			if envVar.Value == "0" {
+				existing.Spec.Template.Spec.Containers[0].Env[envId].Value = "1"
+				update = true
+				return update, nil
+			}
+		}
+	}
+	// Always set env var create CONFIG_REDIS_ASYNC if not present
+	hasListenerWorkers := true
+	hasConfigRedisAsync := true
+	for _, envVar := range existing.Spec.Template.Spec.Containers[0].Env {
+		if existing.Name == "backend-worker" {
+			if envVar.Name == "CONFIG_REDIS_ASYNC" {
+				update = false
+				return update, nil
+			}
+		}
+		if existing.Name == "backend-listener" {
+			if envVar.Name == "LISTENER_WORKERS" {
+				hasListenerWorkers = false
```

Comment:
To me this is a bit confusing to read, if env exists we are assigning false to "hasListenerWorkers" - something small but I would update the naming here or the values so that it is more readable. 

Also as noted above, I would change the flow to:

-    first check if envs are present
-    if not, add them with expected values
-    if yes, ensure that they are in expected state

Reply:
sure can  update the env name. 
As noted above  I believe the order is fine.

Reply:
probably won't need the boolean when separated into different mutators for listener and worker .

Reply:
na still need them.

Reply:
renamed the env vars